### ISSUE TITLE
modify edge_split for splitting a knowledge graph into train/valid/test for entity prediction

### ISSUE THREAD

**Issue by dchang56** [DeepGraphLearning/graphvite#32]
Currently, the random split from edge_split() works only for link prediction.
Could you modify the split to also work for entity prediction?
Thank you!

**Comment by KiddoZhu** [comment]
Good idea. A small problem is that ensuring the entity set are observed in the splits might not be as scalable as previous stream-style random split. We will discuss on how to implement that.

**Comment by dchang56** [comment]
Maybe we could just create a separate test file for entity prediction that's a subset of seen triplets from the original split?

Also, does entity prediction work for the standard datasets for benchmark? And does that mean that the test sets in the benchmark datasets all have seen triplets?

**Comment by KiddoZhu** [comment]
I just checked entity prediction. It works on `FB15k` and `WN18`. So it means their test splits don't have any unobserved entities.

Then we will change `edge_split()` to satisfy that.

**Comment by dchang56** [comment]
Any updates?
